shared/helpers/puppeteer.js

```diff
@@ -111,49 +125,94 @@ async function render(page, componentName, renderOptions, browserOptions) {
-  const error = await page.evaluate(
-    async (selector, exportName, config) => {
-      const namespace = await import('govuk-frontend')
-
-      // Find all matching modules
-      const $modules = document.querySelectorAll(selector)
-
-      try {
-        // Loop and initialise all $modules or use default
-        // selector `null` return value when none found
-        ;($modules.length ? $modules : [null]).forEach(
-          ($module) => new namespace[exportName]($module, config)
-        )
-      } catch ({ name, message }) {
-        return { name, message }
-      }
-    },
-    selector,
-    exportName,
-    browserOptions?.config
-  )
-
-  // Throw Puppeteer errors back to Jest
-  if (error) {
-    throw new Error(
-      `Initialising \`new ${exportName}()\` with example '${exampleName}' threw:` +
-        `\n\t${error.name}: ${error.message}`,
-      { cause: error }
+  try {
+    const error = await page.evaluate(
+      async (selector, exportName, config) => {
+        const namespace = await import('govuk-frontend')
+
+        // Find all matching modules
+        const $modules = document.querySelectorAll(selector)
+
+        try {
+          // Loop and initialise all $modules or use default
+          // selector `null` return value when none found
+          ;($modules.length ? $modules : [null]).forEach(
+            ($module) => new namespace[exportName]($module, config)
+          )
+        } catch ({ name, message }) {
+          return { name, message }
+        }
+      },
+      selector,
+      exportName,
+      browserOptions?.config
     )
+
+    // Throw Puppeteer errors back to Jest
+    if (error) {
+      throw new Error(
+        `Initialising \`new ${exportName}()\` with example '${exampleName}' threw:` +
+          `\n\t${error.name}: ${error.message}`,
+        { cause: error }
+      )
+    }
+
+    await page.evaluateHandle('document.fonts.ready')
+  } finally {
+    // Disable middleware
+    page.off('request', middleware)
+
+    await page.setRequestInterception(false)
   }
 
   return page
 }
 
+/**
+ * Component preview HTTP request middleware
+ *
+ * @param {{ route: URL; body: string; contentType?: string }} options - Route options
+ * @returns {(request: import('puppeteer').HTTPRequest) => Promise<void>} Route middleware
+ */
+const renderMiddleware =
+  ({ route, body, contentType = 'text/html' }) =>
+  async (request) => {
+    if (request.isInterceptResolutionHandled()) {
+      return
+    }
+
+    const { href, pathname, protocol } = new URL(request.url())
+
+    // Return component preview
+    if (href === route.href) {
+      return request.respond({ body, contentType })
+    }
+
+    // Return static assets
+    if (protocol === 'file:' && pathname.startsWith('/assets/')) {
+      return request.respond({
+        body: readFileSync(join(paths.package, `dist/govuk/${pathname}`)),
+        contentType: mime.lookup(pathname) || 'text/plain'
+      })
+    }
```

Comment:
**question** With the paths to our JavaScript, CSS and assets folder being generated by [`pathToFileURL`](https://nodejs.org/api/url.html#urlpathtofileurlpath) in `components.js`, which URLs will have a path starting with `/assets/`?

Reply:
Ah good point, didn't think of those!